panic: interface conversion: dwarf.Type is *dwarf.VoidType, not *dwarf.StructType

Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

1. What version of Delve are you using (`dlv version`)?
Delve Debugger
Version: 0.11.0-alpha
Build: v0.11.0
2. What version of Go are you using? (`go version`)?
1.7.3
3. What operating system and processor architecture are you using?
mac os 10.12.1
4. What did you do?
goland IDE run Debugger & http POST /api/rpm/pricing/lnbase/


runtime: no module data for 0x0
fatal error: no module data
[tango] 2016/12/16 15:28:07 [Debug] logger.go:51 Started POST /api/rpm/pricing/lnbase/ for 172.168.173.5
panic: interface conversion: dwarf.Type is *dwarf.VoidType, not *dwarf.StructType

goroutine 9 [running]:
panic(0x44aab20, 0xc42251a040)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/go/src/runtime/panic.go:500 +0x1a1
github.com/derekparker/delve/proc.mapEvacuated(0xc4235beb60, 0x0)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/proc/variables.go:1379 +0x180
github.com/derekparker/delve/proc.(*mapIterator).nextBucket(0xc4231d4120, 0x8)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/proc/variables.go:1269 +0xa17
github.com/derekparker/delve/proc.(*mapIterator).next(0xc4231d4120, 0xc4231d4120)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/proc/variables.go:1346 +0x49
github.com/derekparker/delve/proc.(*Variable).loadMap(0xc4234e41b0, 0x1, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/proc/variables.go:1159 +0x13b
github.com/derekparker/delve/proc.(*Variable).loadValueInternal(0xc4234e41b0, 0x1, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/proc/variables.go:757 +0xa02
github.com/derekparker/delve/proc.(*Variable).loadValueInternal(0xc4235c6b60, 0x0, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/proc/variables.go:783 +0x3a2
github.com/derekparker/delve/proc.(*Variable).loadValue(0xc4235c6b60, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/proc/variables.go:726 +0x5b
github.com/derekparker/delve/proc.(*EvalScope).extractVariableFromEntry(0xc423055cc0, 0xc42073af00, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff, 0xc4235c9180, 0x0, 0x0)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/proc/variables.go:505 +0x9a
github.com/derekparker/delve/proc.(*EvalScope).variablesByTag(0xc423055cc0, 0x34, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff, 0x4011378, 0x20, 0x44d45e0, ...)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/proc/variables.go:1561 +0x14c
github.com/derekparker/delve/proc.(*EvalScope).LocalVariables(0xc423055cc0, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff, 0x41e20a2, 0x0, 0x0, 0x0, ...)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/proc/variables.go:536 +0x61
github.com/derekparker/delve/service/debugger.(*Debugger).convertStacktrace(0xc420134480, 0xc4200dcd80, 0xb, 0xb, 0xc42398b8f8, 0x0, 0x0, 0xc423081960, 0xc41ffceac9, 0x0)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/service/debugger/debugger.go:739 +0x42c
github.com/derekparker/delve/service/debugger.(*Debugger).Stacktrace(0xc420134480, 0xffffffffffffffff, 0xa, 0xc42398b8f8, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/service/debugger/debugger.go:729 +0x189
github.com/derekparker/delve/service/rpc2.(*RPCServer).Stacktrace(0xc420731ba0, 0xffffffffffffffff, 0xa, 0x1, 0x0, 0xc423055b20, 0x0, 0x0)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/service/rpc2/server.go:143 +0xda
reflect.Value.call(0xc4231e4900, 0xc420132528, 0x13, 0x4527ff9, 0x4, 0xc42398bf10, 0x3, 0x3, 0xc423055b20, 0x0, ...)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/go/src/reflect/value.go:434 +0x5c8
reflect.Value.Call(0xc4231e4900, 0xc420132528, 0x13, 0xc42398bf10, 0x3, 0x3, 0x3, 0x3, 0x0)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/go/src/reflect/value.go:302 +0xa4
github.com/derekparker/delve/service/rpccommon.(*ServerImpl).serveJSONCodec(0xc420130410, 0x5ea0678, 0xc420140010)
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/service/rpccommon/server.go:288 +0x68f
created by github.com/derekparker/delve/service/rpccommon.(*ServerImpl).Run.func1
	/Users/builduser/buildAgent/work/b97f8195591f35ac/gopath/src/github.com/derekparker/delve/service/rpccommon/server.go:146 +0xdc

Process finished with exit code 2

Hello, 
same problem here, with same versions of @jjjachyty. Console output is:

```
panic: interface conversion: dwarf.Type is *dwarf.VoidType, not *dwarf.StructType

goroutine 9 [running]:
panic(0x44b3540, 0xc423ee4240)
	/usr/local/Cellar/go/1.7.4_1/libexec/src/runtime/panic.go:500 +0x1a1
github.com/derekparker/delve/proc.(*mapIterator).nextBucket(0xc423d1e1e0, 0x8)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/proc/variables.go:1301 +0x80c
github.com/derekparker/delve/proc.(*mapIterator).next(0xc423d1e1e0, 0xc423d1e1e0)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/proc/variables.go:1346 +0x49
github.com/derekparker/delve/proc.(*Variable).loadMap(0xc423edeb60, 0x0, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/proc/variables.go:1159 +0x13b
github.com/derekparker/delve/proc.(*Variable).loadValueInternal(0xc423edeb60, 0x0, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/proc/variables.go:757 +0xa02
github.com/derekparker/delve/proc.(*Variable).loadValue(0xc423edeb60, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/proc/variables.go:726 +0x5b
github.com/derekparker/delve/proc.(*EvalScope).extractVariableFromEntry(0xc421e88160, 0xc4229e3fb0, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff, 0xc423ed6000, 0x0, 0x0)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/proc/variables.go:505 +0x9a
github.com/derekparker/delve/proc.(*EvalScope).variablesByTag(0xc421e88160, 0x34, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff, 0x4011c08, 0x20, 0x44dd6e0, ...)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/proc/variables.go:1561 +0x14c
github.com/derekparker/delve/proc.(*EvalScope).LocalVariables(0xc421e88160, 0x1, 0x1, 0x40, 0x40, 0xffffffffffffffff, 0xc421e7ac10, 0x2, 0x2, 0x0, ...)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/proc/variables.go:536 +0x61
github.com/derekparker/delve/service/debugger.(*Debugger).convertStacktrace(0xc4200e1700, 0xc4200b2d80, 0x9, 0xb, 0xc421cfd8e8, 0x0, 0x0, 0xc423190fb8, 0xc41fe49c6d, 0x0)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/service/debugger/debugger.go:754 +0x42c
github.com/derekparker/delve/service/debugger.(*Debugger).Stacktrace(0xc4200e1700, 0xffffffffffffffff, 0xa, 0xc421cfd8e8, 0x0, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/service/debugger/debugger.go:744 +0x189
github.com/derekparker/delve/service/rpc2.(*RPCServer).Stacktrace(0xc42366b7b0, 0xffffffffffffffff, 0xa, 0x1, 0x0, 0xc42317b160, 0x0, 0x0)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/service/rpc2/server.go:159 +0xda
reflect.Value.call(0xc42368ad20, 0xc420150528, 0x13, 0x45315e8, 0x4, 0xc421cfdf00, 0x3, 0x3, 0xc42317b160, 0x0, ...)
	/usr/local/Cellar/go/1.7.4_1/libexec/src/reflect/value.go:434 +0x5c8
reflect.Value.Call(0xc42368ad20, 0xc420150528, 0x13, 0xc421cfdf00, 0x3, 0x3, 0x3, 0x3, 0x0)
	/usr/local/Cellar/go/1.7.4_1/libexec/src/reflect/value.go:302 +0xa4
github.com/derekparker/delve/service/rpccommon.(*ServerImpl).serveJSONCodec(0xc42001ae60, 0x6409380, 0xc42013a018)
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/service/rpccommon/server.go:288 +0x68f
created by github.com/derekparker/delve/service/rpccommon.(*ServerImpl).Run.func1
	/private/tmp/delve-20161228-5164-ehs0o1/src/github.com/derekparker/delve/service/rpccommon/server.go:146 +0xdc

Process finished with exit code 2
```

Here the out.txt in gzipped format:

[out.txt.gz](https://github.com/derekparker/delve/files/698109/out.txt.gz)

Regards

--- a/proc/variables.go
+++ b/proc/variables.go
@@ -1237,8 +1237,17 @@ func (v *Variable) mapIterator() *mapIterator {
 		}
 	}
 
+	if it.buckets.Kind != reflect.Struct || it.oldbuckets.Kind != reflect.Struct {
+		v.Unreadable = mapBucketsNotStructErr
+		return nil
+	}
+
 	return it
 }
+
+var mapBucketContentsNotArrayErr = errors.New("malformed map type: keys, values or tophash of a bucket is not an array")
+var mapBucketContentsInconsistentLenErr = errors.New("malformed map type: inconsistent array length in bucket")
+var mapBucketsNotStructErr = errors.New("malformed map type: buckets, oldbuckets or overflow field not a struct")
 
 func (it *mapIterator) nextBucket() bool {
 	if it.overflow != nil && it.overflow.Addr > 0 {
@@ -1328,12 +1337,17 @@ func (it *mapIterator) nextBucket() bool {
 	}
 
 	if it.tophashes.Kind != reflect.Array || it.keys.Kind != reflect.Array || it.values.Kind != reflect.Array {
-		it.v.Unreadable = fmt.Errorf("malformed map type: keys, values or tophash of a bucket is not an array")
+		it.v.Unreadable = mapBucketContentsNotArrayErr
 		return false
 	}
 
 	if it.tophashes.Len != it.keys.Len || it.tophashes.Len != it.values.Len {
-		it.v.Unreadable = fmt.Errorf("malformed map type: inconsistent array length in bucket")
+		it.v.Unreadable = mapBucketContentsInconsistentLenErr
+		return false
+	}
+
+	if it.overflow.Kind != reflect.Struct {
+		it.v.Unreadable = mapBucketsNotStructErr
 		return false
 	}
 

